store/sqlstore/team_store.go

```diff
@@ -491,18 +491,17 @@ func (s SqlTeamStore) getTeamMembersWithSchemeSelectQuery() sq.SelectBuilder {
 		LeftJoin("Schemes TeamScheme ON Teams.SchemeId = TeamScheme.Id")
 }
 
-func (s SqlTeamStore) SaveMember(member *model.TeamMember, maxUsersPerTeam int) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		defer s.InvalidateAllTeamIdsForUser(member.UserId)
-		if result.Err = member.IsValid(); result.Err != nil {
-			return
-		}
+func (s SqlTeamStore) SaveMember(member *model.TeamMember, maxUsersPerTeam int) (*model.TeamMember, *model.AppError) {
+	defer s.InvalidateAllTeamIdsForUser(member.UserId)
+	if err := member.IsValid(); err != nil {
+		return nil, err
+	}
 
-		dbMember := NewTeamMemberFromModel(member)
+	dbMember := NewTeamMemberFromModel(member)
 
-		if maxUsersPerTeam >= 0 {
-			count, err := s.GetMaster().SelectInt(
-				`SELECT
+	if maxUsersPerTeam >= 0 {
+		count, err := s.GetMaster().SelectInt(
+			`SELECT
 					COUNT(0)
```

Comment:
do you mind to indent accordingly the rest of the query text (1 tab less)?

Reply:
sure